### PR TITLE
docs(ref): Simplify check-cfg build.rs docs

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

After looking over the rust documentation updates, decided to look over Cargo's build.rs documentation.

- Removed link to the blog post as that is not evergreen documentation
- Collapsed down the example to put more emphasis on the more complete example

### CODE REVIEW

**Review by weihanglo — APPROVED** [review]
Thanks for the simplification.

File: src/doc/src/reference/build-scripts.md
Comment (weihanglo) [inline src/doc/src/reference/build-scripts.md:1]: Should we have a link to the new page in rustc book?

https://github.com/rust-lang/rust/pull/125219

Comment (epage) [reply src/doc/src/reference/build-scripts.md:1]: Sounds like that page is a short-term hack

> However, I'd rather keep Cargo-specific explainers that are as long as this one out of the rustc book

(which I agree with)

so I'm a bit hesitant linking to a page that will soon be broken.

Comment (weihanglo) [reply src/doc/src/reference/build-scripts.md:1]: Yes, but docs are designed for people to read. They are less useful if there is no route to them. We also haven't put a date for removing that, so I think it's better to link to that within the same version. The linkchcker should be able to detect breakages when the page is removed.

Comment (epage) [reply src/doc/src/reference/build-scripts.md:1]: Looking more closely at each piece of documentation, I'm not seeing much reason to link to the cargo-specific page.  It just duplicates what we are already trying to cover.  Instead, we are missing a link to the parent page which I added.

Comment (epage) [reply src/doc/src/reference/build-scripts.md:1]: (the cargo page is more of a starting point from the lint output)

**Pull request merged**